Ghost name QoL changes

coiax opened the pull request:
:cl: coiax
rscadd: Ghosts now have the "Restore Character Name" verb, which will set
their ghost appearance and dead chat name to their character preferences.
bugfix: Mob spawners now give any generated names to the mind of the
spawned mob, meaning ghosts will have the name they had in life.
bugfix: Fixes interaction with envy's knife and lavaland spawns.
/:cl:

Fixes #17842.

coiax committed:
Ghost name QoL changes

:cl: coiax
rscadd: Ghosts now have the "Restore Character Name" verb, which will set
their ghost appearance and name to their character preferences.
bugfix: Mob spawners now give any generated names to the mind of the
spawned mob, meaning ghosts will have the name they had in life.
the outsiders as they see fit.
/:cl:

...

fsdjkafklasdfljk

RemieRichards commented on code/modules/mob/dead/observer/observer.dm:
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 
+/mob/dead/observer/verb/restore_ghost_apperance()
+	set name = "Restore Character Name"
+	set desc = "Sets your ghost name to the name of your roundstart character."
```
>but it doesn't do just name

RemieRichards commented on code/modules/mob/dead/observer/observer.dm:
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		data_huds_on = 1
 
+/mob/dead/observer/verb/restore_ghost_apperance()
+	set name = "Restore Character Name"
```
but it doesn't do just name

coiax replied:
THINK OF THE GHOST HAIR

RemieRichards commented on code/modules/mob/dead/observer/observer.dm:
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
+
+/mob/dead/observer/proc/set_ghost_appearance()
+	if(!client)
+		world << "NO CLIENT"
```
debug text

RemieRichards commented on code/modules/mob/dead/observer/observer.dm:
```diff
@@ -590,6 +590,40 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
+		world << "NO CLIENT"
+		return
+	if(!client.prefs)
+		world << "NO PREFS"
```
debug text

optimumtact commented:
Feature freeze

optimumtact commented:
Reopened with the understanding the ash walker tweak will be removed

Shadowlight213 commented:
:-1: this is going to be a pain in the ass for administration, if someone says X griefed me, and admins look for x's ghost, but Surprise! X has changed his name to something completely different and now admins have to go dig up logs just to figure out who the they were.

coiax committed:
Made Restore Ghost Character affect deadchat only

Instead of altering the name of the ghost mob and mind, now it just lets
you babble in deadchat using your regular name with a special var.

coiax commented:
@Shadowlight213 what about now? Only affects the name presented to deadchat, doesn't alter any of the vars of the ghost mob or mind.

Shadowlight213 commented:
better
